File: demo_py/demo_py/person.py
import rclpy
from rclpy.node import Node

class PersonNode(Node):
    def __init__(self,name,age):
        super().__init__('person_node')
        self.name = name
        self.age = age

    def print_info(self):
        self.get_logger().info(f"Name: {self.name}, Age: {self.age}")


def main():
    rclpy.init()
    person_node = PersonNode("wjz", 20)
    person_node.print_info()
    rclpy.spin(person_node)
    rclpy.shutdown()

File: demo_cpp/src/person.cpp
#include "rclcpp/rclcpp.hpp"
#include <string>


class PersonNode : public rclcpp::Node 
{
private:
    std::string name_;
    int age_;

public:
    PersonNode(const std::string &node_name, const std::string &name, int age)
        : Node(node_name)
    {
        this->name_ = name;
        this->age_ = age;
    };
    void print_info()
    {
        RCLCPP_INFO(this->get_logger(), "Name: %s, Age: %d", this->name_.c_str(), this->age_);
    };
};

int main(int argc, char const *argv[])
{
    rclcpp::init(argc,argv);
    auto node = std::make_shared<PersonNode>("Person_cpp_node","wjz",18);
    node->print_info();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

